lib/view_component/compiler.rb

```diff
@@ -163,9 +163,8 @@ def matching_views_in_source_location
       # end
       #
       # Without this, `MyOtherComponent` will not look for `my_component/my_other_component.html.erb`
-      nested_component_files = if component_class.name.include?("::")
-        nested_component_path = component_class.name.deconstantize.underscore
-        Dir["#{directory}/#{nested_component_path}/#{component_name}.*{#{extensions}}"]
+      nested_component_files = if component_class.name.include?("::") && component_name != filename
```

Comment:
Adding this additional conditional seemed worth doing, to avoid doing this check when the names match up, in which case I don't believe we could possibly be dealing with a nested component.

Reply:
I think this seems fine since there shouldn't ever be a case where the component has a nested component with the same name.